shared-libs/contacts/test/unit/places.spec.js

```diff
@@ -338,22 +338,47 @@ describe('places controller', () => {
       };
       sinon.stub(people, 'getOrCreatePerson').resolves({
         _id: 'qwe',
+        _rev: '1',
         name: 'Jim',
         type: 'person'
       });
-      fetchHydratedDoc.resolves({
-        _id: 'ad06d137',
-        name: 'CHP Branch One',
-        type: 'district_hospital'
-      });
       db.medic.post.callsFake(doc => {
-        chai.expect(doc.contact._id).to.equal('qwe');
-        chai.expect(doc.contact.name).to.equal(undefined); // minified
-        chai.expect(doc.contact.type).to.equal(undefined); // minified
-        return Promise.resolve({id: 'ghi'});
+        if (doc.name === 'CHP Family') {
+          return Promise.resolve({ id: 'hc', rev: '1' });
+        }
+        if (doc.name === 'CHP Branch One') {
+          return Promise.resolve({ id: 'ad06d137' });
+        }
+      });
```

Comment:
Since i know the order of the calls, can i just use `db.medic.post.onCall(0).callsFake` instead?

Reply:
The problem with using the order of calls is that you end up with testing an implementation detail, and if the calls change order you also need to change the test. Also, from a reader's perspective, it's easier to follow an arguments -> response correlation rather than checking the order of calls, the test will make sense sooner.

Reply:
makes sense, updated to use sinon matcher func